### PR TITLE
remove ci step that publishes docs to github pages and auto publish docs to s3 during release

### PR DESCRIPTION
**Pull request by grant-Kolena**
### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -49,15 +49,10 @@ jobs:
...
         run: poetry build --format=sdist
 
-      - name: Build 'kolena' documentation and push to GitHub Pages
+      - name: Publish 'kolena' documentation to production (S3)
         run: |
-          git config user.name 'kolenabot'
-          git config user.email 'bot@kolena.io'
-          ./docs/setup_insiders.sh
-          poetry run mkdocs gh-deploy --verbose --strict --remote-branch release/docs --config-file mkdocs.insiders.yml
-        env:
```
Comment (gordonhart) [inline .github/workflows/release.yml:58]: Are these vars not still necessary? We still have DD RUM plugged into the docs.

Comment (grant-Kolena) [reply .github/workflows/release.yml:58]: We still have them in the mkdocs config [here](https://github.com/kolenaIO/kolena/blob/de651a6f7e4ad548c5bd0796da3df60d41e010dd/mkdocs.yml#L215-L217)

Comment (gordonhart) [reply .github/workflows/release.yml:58]: If they're not mounted in these CI steps, how are they getting populated? I see `blank` tokens on docs.kolena.io currently.

Comment (grant-Kolena) [reply .github/workflows/release.yml:58]: How are they used? can we also remove that?

Comment (grant-Kolena) [reply .github/workflows/release.yml:58]: seems this is related to datadog

Comment (grant-Kolena) [reply .github/workflows/release.yml:58]: asking around in general channel

Comment (grant-Kolena) [reply .github/workflows/release.yml:58]: for completeness, the doc for the dd rum browser setup can be found [here](https://docs.datadoghq.com/real_user_monitoring/browser/#cdn-sync), we use the CDN sync approach (our [code](https://github.com/kolenaIO/kolena/blob/trunk/docs/overrides/main.html#L16)). Just need to set the variables before building the docs

**Review by gordonhart — APPROVED** [review]

**Pull request merged**